src/integration/user-registration.integration.test.ts

```diff
@@ -0,0 +1,683 @@
+          .expect(200);
+
+        expect(response.body).toHaveProperty("user");
+        expect(response.body.user.email).toBe(userData.email);
+      });
+
+      it("should accept various valid email formats", async () => {
+        const validEmails = [
+          `user.name.${Date.now()}@example.com`,
+          `user+tag${Date.now()}@example.co.uk`,
+          `first.last${Date.now()}@company-name.com`,
+        ];
+
+        for (const email of validEmails) {
+          const response = await request(app)
+            .post("/api/auth/sign-up/email")
+            .send({
+              email,
+              password: validPassword,
+            })
+            .set("Content-Type", "application/json");
+
+          expect(response.status).toBe(200);
+          expect(response.body.user.email).toBe(email);
+        }
+      });
+
+      it("should accept various valid password formats", async () => {
+        const validPasswords = [
+          "Password123",
+          "MyP@ssw0rd",
+          "Test1234!@#",
+          "ValidP4ss",
+        ];
+
+        for (const password of validPasswords) {
+          const response = await request(app)
+            .post("/api/auth/sign-up/email")
+            .send({
+              email: generateUniqueEmail(),
+              password,
+            })
+            .set("Content-Type", "application/json");
+
+          expect(response.status).toBe(200);
+        }
+      });
+    });
+
+    describe("Response Structure Validation", () => {
+      it("should return user object with all expected fields", async () => {
+        const userData = createValidRegistrationData();
+
+        const response = await request(app)
+          .post("/api/auth/sign-up/email")
+          .send(userData)
+          .set("Content-Type", "application/json")
+          .expect(200);
+
+        const { user } = response.body;
+
+        // Check all expected fields exist
+        const expectedFields = [
+          "id",
+          "email",
+          "name",
+          "emailVerified",
+          "createdAt",
+          "updatedAt",
+        ];
+        expectedFields.forEach((field) => {
+          expect(user).toHaveProperty(field);
+        });
+
+        // Validate timestamps are valid ISO strings
+        expect(new Date(user.createdAt).toISOString()).toBe(user.createdAt);
+        expect(new Date(user.updatedAt).toISOString()).toBe(user.updatedAt);
+      });
+
+      it("should never include password in response", async () => {
+        const userData = createValidRegistrationData();
+
+        const response = await request(app)
+          .post("/api/auth/sign-up/email")
+          .send(userData)
+          .set("Content-Type", "application/json")
+          .expect(200);
+
+        // Password should not be in the response at all
+        expect(response.body.user).not.toHaveProperty("password");
+        expect(JSON.stringify(response.body)).not.toContain(userData.password);
+      });
```

Comment:
I'd move the assertions from line 179-180 to the test above and then delete this test

It can be covered under that 1st one :)